File: diff_panels.py
import difflib

file1 = r'c:\Users\iWexort\Documents\Github\mirzabot-main\temp_mirzabot\panels.php'
file2 = r'c:\Users\iWexort\Documents\Github\mirzabot-main\panels.php'

with open(file1, 'r', encoding='utf-8') as f1, open(file2, 'r', encoding='utf-8') as f2:
    lines1 = f1.readlines()
    lines2 = f2.readlines()

diff = difflib.unified_diff(lines1, lines2, fromfile=file1, tofile=file2, n=3)

with open(r'c:\Users\iWexort\Documents\Github\mirzabot-main\diff_panels.txt', 'w', encoding='utf-8') as out:
    out.writelines(diff)

print("Diff complete. Written to diff_panels.txt.")

File: diff_panels.php
<?php
$file1 = 'temp_mirzabot/panels.php';
$file2 = 'panels.php';

if (!file_exists($file1) || !file_exists($file2)) {
    die("One of the files does not exist.\n");
}

$lines1 = file($file1, FILE_IGNORE_NEW_LINES);
$lines2 = file($file2, FILE_IGNORE_NEW_LINES);

$diff = [];
$len1 = count($lines1);
$len2 = count($lines2);
$max = max($len1, $len2);

for ($i = 0; $i < $max; $i++) {
    $l1 = isset($lines1[$i]) ? $lines1[$i] : '[EOF]';
    $l2 = isset($lines2[$i]) ? $lines2[$i] : '[EOF]';
    if ($l1 !== $l2) {
        $diff[] = "Line " . ($i + 1) . ":\n- " . $l1 . "\n+ " . $l2 . "\n";
    }
}

file_put_contents('diff_panels.txt', implode("\n", $diff));
echo "Diff complete, written to diff_panels.txt. Total diff size: " . count($diff) . " blocks.\n";
?>
